components/zeekendpointsecurity/tests/endpointsecurityconsumer.cpp

```diff
@@ -60,4 +60,199 @@ TEST_CASE("Event header initialization", "[ZeekEndpointConsumer]") {
   REQUIRE(header.path == kDummyPath);
   REQUIRE(header.cdhash == "a0a1a2a3a4a5a6a7a8a9aaabacadaeafb0b1b2b3");
 }
+
+TEST_CASE("Event header initialization for Open event",
+          "[ZeekEndpointConsumer]") {
+  es_process_t process = {};
+  process.ppid = 10U;
+  process.original_ppid = 11U;
+
+  for (auto i = 0U; i < 8U; ++i) {
+    process.audit_token.val[i] = 0xF0U + i;
+  }
+
+  process.is_platform_binary = 1U;
+
+  for (auto i = 0U; i < sizeof(process.cdhash); ++i) {
+    process.cdhash[i] = 0xA0U + i;
+  }
+
+  const char *kDummySigningIdentifier = "SigningID";
+  process.signing_id.data = kDummySigningIdentifier;
+  process.signing_id.length = std::strlen(kDummySigningIdentifier);
+
+  const char *kDummyTeamIdentifier = "TeamID";
+  process.team_id.data = kDummyTeamIdentifier;
+  process.team_id.length = std::strlen(kDummyTeamIdentifier);
+
+  const char *kDummyPath = "/path/to/application";
+  es_file_t executable = {};
+  executable.path.data = kDummyPath;
+  executable.path.length = std::strlen(kDummyPath);
+
+  process.executable = &executable;
+
+  es_message_t es_message = {};
+  es_message.event_type = ES_EVENT_TYPE_NOTIFY_OPEN;
+  es_message.process = &process;
+
+  const char *kDummyFilePath = "/path/to/file";
+  es_file_t file = {};
+  file.path.data = kDummyFilePath;
+  file.path.length = std::strlen(kDummyFilePath);
+
+  es_message.event.open.file = &file;
+
+  IEndpointSecurityConsumer::Event::Header header;
+  auto status =
+      EndpointSecurityConsumer::initializeEventHeader(header, es_message);
+
+  REQUIRE(status.succeeded());
+
+  REQUIRE(header.timestamp != 0U);
```

Comment:
While we want to `REQUIRE(status.succeeded())` as a precondition of any later checks, all others here should be `CHECK`s.